PAY-6182: Adapt docs to Payment Services drop-in V2 (#433)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/initialization.mdx

---

import OptionsTable from '@components/OptionsTable.astro';

The Payment Services drop-in component initializer provides options for setting the backend endpoint and configuring language definitions.

## Configuration options

The Payment Services component initializer accepts the following configuration options:

<OptionsTable
  options={[
    ['Option', 'Type', 'Req?', 'Description'],
    ['apiUrl', 'string', 'Yes', 'The Adobe Commerce GraphQL endpoint URL, such as "https://example.com/graphql".'],
    ['getCustomerToken', 'function', 'Yes', 'The Payment Services drop-in component may send GraphQL requests on behalf of the shopper. This requires GraphQL authorization, which can be performed using authorization tokens or session cookies. For token-based authorization, the "getCustomerToken" function should return a customer token as a string, or null for guest customers. For session-based authorization, the "getCustomerToken" property must explicitly be set to "null".'],
    ['langDefinitions', 'object', 'No', 'Provides language definitions for internationalization (i18n).'],
  ]}
/>

## Initialization is asynchronous

> The Payment Services drop-in initializer is completely asynchronous. Even if mounted immediately, the drop-in will initialize in the background and notify initialization updates through custom events.

Reviewer: ```suggestion
The Payment Services drop-in initializer is fully asynchronous. Even if mounted immediately, it initializes in the background and emits initialization updates via custom events.
```

## Payment method availability

> The Payment Services drop-in will emit a custom `payment-services/method-available` event for every payment method that is determined to be available during the asynchronous initialization process. The event payload is a single string containing the payment method code.

Reviewer: ```suggestion
During asynchronous initialization, the Payment Services drop-in emits a `payment-services/method-available` event for each available payment method. The event payload is a single string containing the payment method code.
```
